src/Flexible.php

```diff
@@ -139,12 +139,14 @@ public function confirmRemove($label = '', $yes = 'Delete', $no = 'Cancel')
     /**
      * Set the field's resolver
      *
-     * @param string $classname
+     * @param mixed $resolver
      * @return $this
      */
-    public function resolver($classname)
+    public function resolver($resolver)
     {
-        $resolver = new $classname();
+        if(!($resolver instanceof ResolverInterface)) {
```

Comment:
Here we should test if the `$resolver` argument is a valid classname string instead of testing if it is not a ResolverInterface (that check is done later, on line 151). So, instead of your condition, I would use :

```php
if(is_string($resolver) && is_a($resolver, ResolverInterface::class, true))
```

For more information, check the [is_a()](https://www.php.net/manual/fr/function.is-a.php) documentation.

Reply:
Yes, this is a solid check as you suggested, thanks. Already pushed to your repo! Btw, the pattern I followed for the argument check is based on the `$layout` argument in the `addLayout` method which was accepted in #90

Reply:
Fair point. The condition added in #90 should be updated as-well. Thank you !